### ISSUE TITLE
Permission on dir not set (chmod not effective)

### ISSUE THREAD

**Issue by ngauje** [GoogleContainerTools/kaniko#1185]
**Actual behavior**
In a Dockerfile with `RUN chmod -R 775 /var`, the permission is not set on the directory. Even worse, if the /var was already set to 775, kaniko unset the perm! Without `chmod` the perms are OK 

No multistage or volumes used, very simple Dockerfiles used (see below).

Bad perms with kaniko :
```
docker run --rm myregistry/build-with-kaniko:latest ls -ld /var
drwxr-xr-x    1 root     root            41 Apr  8 23:16 /var
```

Correct perms with docker:
```
docker run --rm myregistry/build-with-docker:latest ls -ld /var                                                                 
drwxrwxr-x    1 root     root           137 Apr  8 23:14 /var
```

**Expected behavior**
A /var directory with correct permission (775 in my example)

**To Reproduce**
1. Build a baseimage (with docker) with the following Dockerfile:
```
FROM alpine:edge

RUN mkdir -p /var/www/app \
 && chgrp -R 0 /var \
 && chmod -R g+rwX /var

RUN ls -ld /var
```
2. Build another image based on the 1st one with Kaniko. Dockerfile:
`echo 'YOLO' > file1.txt`

```
FROM myregistry/baseimage:latest

COPY file1.txt /var/www/app/file1.txt

RUN chmod -R 775 /var
RUN ls -ld /var
```
3. Build output:
`/kaniko/executor --context . --dockerfile Dockerfile --destination=myregistry/test-kaniko-bug:latest`
```
[...]
INFO[0019] args: [-c chmod -R 775 /var]                 
INFO[0019] Taking snapshot of full filesystem...        
INFO[0019] Resolving paths                              
INFO[0019] RUN ls -ld /var                              
INFO[0019] cmd: /bin/sh                                 
INFO[0019] args: [-c ls -ld /var]                       
drwxrwxr-x   13 root     root           137 Apr  8 23:31 /var
[...]
```
At build stage perms are OK
4. Run the container in Docker:
```
docker run --rm myregistry/test-kaniko-bug:latest ls -ld /var
drwxr-xr-x    1 root     root            41 Apr  8 23:33 /var
```
Perms are not good

5. If I comment out the `RUN chmod -R 775 /var` it works !

**Additional Information**
 - Kaniko Image (fully qualified with digest) : 
`gcr.io/kaniko-project/executor   debug               2ec307dcf7f5`
 - Same problem with debug-edge
 - Build with Docker is OK:
`docker build -t newimage .`
```
Sending build context to Docker daemon  3.072kB
[...]
Step 3/4 : RUN chmod -R 775 /var
 ---> Running in aedc94aae680
Removing intermediate container aedc94aae680
 ---> 9bc8cf64d38d
Step 4/4 : RUN ls -ld /var
 ---> Running in 6e4079bee7da
drwxrwxr-x    1 root     root           137 Apr  8 23:14 /var
Removing intermediate container 6e4079bee7da
 ---> 48ad15e5232c
Successfully built 48ad15e5232c
Successfully tagged newimage:latest
```
```
docker run --rm newimage:latest ls -ld /var                                                                 
drwxrwxr-x    1 root     root           137 Apr  8 23:14 /var
```
 - `Docker version 19.03.4, build 9013bf583a`

 
 **Triage Notes for the Maintainers**
 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [x] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by tomprince** [comment]
This looks like a duplicate of  #1163 (fix in  #1166).

**Comment by ngauje** [comment]
Fixed in v0.21.0  (maybe debug-v0.20.0 ...)
Thanks!
